api/contrib/envoy/extensions/filters/http/golang/v3alpha/golang.proto

```diff
@@ -0,0 +1,217 @@
+//                     - name: envoy.filters.http.golang
+//                       typed_config:
+//                         "@type": type.googleapis.com/envoy.extensions.filters.http.golang.v3alpha.Config
+//                         library_id: auth-id
+//                         library_path: "/home/admin/envoy/go-plugins/libauth.so"
+//                         plugine_name: auth
+//                         plugin_config:
+//                           "@type": type.googleapis.com/envoy.extensions.filters.http.golang.plugins.auth.v3.Config
+//                           xxx1: xx1
+//                           xxx2: xx2
+//                     - name: envoy.filters.http.router
+// [#next-free-field: 6]
+message Config {
+  enum MergePolicy {
+    MERGE_VIRTUALHOST_ROUTER_FILTER = 0;
+    MERGE_VIRTUALHOST_ROUTER = 1;
+    OVERRIDE = 3;
+  }
+
+  // library_id is a unique ID for a dynamic library file, must be unique globally.
+  string library_id = 1 [(validate.rules).string = {min_len: 1}];
+
+  // Dynamic library implementing the interface of
+  // ``StreamFilter <contrib/golang/filters/http/source/go/pkg/api.StreamFilter>``.
+  // [#comment:TODO(wangfakang): Support for downloading libraries from remote repositories.]
+  string library_path = 2 [(validate.rules).string = {min_len: 1}];
+
+  // plugin_name is the name of the go plugin, which needs to be consistent with the name
+  // registered in http::RegisterHttpFilterConfigFactory.
+  string plugin_name = 3 [(validate.rules).string = {min_bytes: 1}];
+
+  // plugin_config is the configuration of the go plugin, note that this configuration is
+  // only parsed in the go plugin.
+  google.protobuf.Any plugin_config = 4;
+
+  // merge_policy is the merge policy configured by the go plugin.
+  // go plugin configuration supports three dimensions: the virtual host’s typed_per_filter_config,
+  // the route’s typed_per_filter_config or filter's config.
+  // The meanings are as follows:
+  // MERGE_VIRTUALHOST_ROUTER_FILTER: pass all configuration into go plugin.
+  // MERGE_VIRTUALHOST_ROUTER: pass Virtual-Host and Router configuration into go plugin.
+  // OVERRIDE: override according to Router > Virtual_host > Filter priority and pass the
+  // configuration to the go plugin.
+  MergePolicy merge_policy = 5 [(validate.rules).enum = {defined_only: true}];
+}
+
+message RouterPlugin {
+  // The extension_plugin_options field is used to provide extension options for plugin.
+  // Example
+  //
+  // .. code-block:: yaml
+  //
+  //   typed_per_filter_config:
+  //     envoy.filters.http.golang:
+  //       "@type": type.googleapis.com/envoy.extensions.filters.http.golang.v3alpha.ConfigsPerRoute
+  //       plugins_config:
+  //         plugin1:
+  //          extension_plugin_options:
+  //            disabled: true
+  //          config:
+  //            "@type": type.googleapis.com/golang.http.plugin1
+  //            xxx: xxx
+  google.protobuf.Struct extension_plugin_options = 1;
```

Comment:
Let's please just make this fully typed. It's much easier to document and understand. As they API is alpha we can always change it later. Thank you.

/wait

Reply:
Done.